surelock/src/main/java/com/smashingboxes/surelock/Surelock.java

```diff
@@ -497,4 +459,142 @@ private void setEncryptionIv(byte[] encryptionIv) {
+            this.fingerprintDialogFragmentTag = fingerprintDialogFragmentTag;
+        }
+
+        /**
+         * Indicates that fingerprint login should be prompted using the SurelockDefaultDialog
+         * class. This is a fullscreen dialog that can be styled to match an app's theme.
+         *
+         * @param styleId The style resource file to be used for styling the dialog
+         * @return This SurelockDialogBuilder to allow for method chaining
+         */
+        public SurelockDialogBuilder withDefaultDialog(@StyleRes int styleId) {
+            useDefault = true;
+            surelockDialog = null;
+            this.styleId = styleId;
+            return this;
+        }
+
+        /**
+         * Indicates that fingerprint login should be prompted using the SurelockMaterialDialog.
+         * This dialog follows Material Design guidelines.
+         *
+         * @return This SurelockDialogBuilder to allow for method chaining
+         */
+        public SurelockDialogBuilder withMaterialDialog() {
+            useDefault = false;
+            surelockDialog = null;
+            return this;
+        }
+
+        /**
+         * Indicates that fingerprint login should be prompted using the given dialog.
+         *
+         * @param surelockDialog The custom dialog to use for fingerprint login
+         * @return This SurelockDialogBuilder to allow for method chaining
+         */
+        public SurelockDialogBuilder withCustomDialog(@NonNull SurelockFragment surelockDialog) {
+            this.surelockDialog = surelockDialog;
+            return this;
+        }
+
+        /**
+         * Indicates the alias to use for the keystore when using fingerprint login. This method
+         * MUST be called before enrolling and logging in.
+         *
+         * @param keystoreAlias The keystore alias to use
+         * @return This SurelockDialogBuilder to allow for method chaining
+         */
+        public SurelockDialogBuilder withKeystoreAlias(@NonNull String keystoreAlias) {
+            this.keystoreAlias = keystoreAlias;
+            return this;
+        }
+
+        /**
+         * Indicates the SurelockStorage instance to use with fingerprint login. This method MUST
+         * be called before enrolling and logging in.
+         *
+         * @param storage The SurelockStorage instance to use
+         * @return This SurelockDialogBuilder to allow for method chaining
+         */
+        public SurelockDialogBuilder withSurelockStorage(@NonNull SurelockStorage storage) {
+            this.storage = storage;
+            return this;
+        }
+
+        /**
+         * Enroll a fingerprint and encrypt the given value, if desired
+         *
+         * @param valueToEncrypt The optional value to be encrypted
+         * @param context        The context used to create the Surelock instance
+         */
+        public void enrollFingerprintAndStore(Context context, @Nullable byte[] valueToEncrypt) {
```

Comment:
Ah I see. I guess I misunderstood 😅 I can change it to return a Surelock instance. That's clearer anyway since a Builder in a class should return the class itself 🤦‍♂️ 

If I do that, should I make the constructor and `initialize()` method private? That way users have to use the builder to get an instance. It looks like Picasso made theirs package local.